src/directives/leaflet.js

```diff
@@ -24,9 +23,9 @@ angular.module("leaflet-directive", []).directive('leaflet',
         transclude: true,
         template: '<div class="angular-leaflet-map"><div ng-transclude></div></div>',
         controller: function ($scope) {
-            _leafletMap = $q.defer();
+            this._leafletMap = $q.defer();
```

Comment:
Sorry but I don't understand the problem with using it as an attribute. What cleanup are you talking about?

Moreover, tell me if I'm wrong, but if I have 2 maps on a page, the _leafletMap should be unique to each map, shouldn't it?